helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java

```diff
@@ -354,12 +354,14 @@ public void testMaxPartitionLimitation() throws Exception {
       String limitedResourceName = null;
       int i = 0;
       for (String stateModel : _testModels) {
-        String db = "Test-DB-" + i++;
+        String db = "Test-DB-" + TestHelper.getTestMethodName() + i++;
         createResourceWithWagedRebalance(CLUSTER_NAME, db, stateModel, PARTITIONS, _replica,
             _replica);
         if (i == 1) {
-          // The limited resource has additional limitation, so even the other resources can be assigned
-          // later, this resource will still be blocked by the max partition limitation.
+          // The limited resource has additional limitation.
+          // The other resources could have been assigned in theory if the WAGED rebalancer were
+          // not used.
+          // However, with the WAGED rebalancer, this restricted resource will block the other ones.
```

Comment:
1. What do you exactly mean by "restricted resource"? Define?
2. Why would it block other ones? Clarify?

Some of these comments are confusing, let's make them clearer.

Reply:
Here the test logic is to verify if the resource level limitation works. When it works fine, the rebalancer won't be able to complete calculating. That's pretty much what the comment says.

I personally think this comment is good enough. In general, I don't like documenting too much about the feature in a test. If you are still concerning, let's forward this to a 3rd Helix dev for additional opinion : )